### PR TITLE
API for unsharing a shared customlist

### PR DESCRIPTION
**Pull request by RishiDiwanTT**
## Description
Unsharing is only allowed if the shared list is not in use by any other library 
To unshare do an HTTP DELETE on the share endpoint

## Motivation and Context
Currently, there is no way to unshare a list once it has been shared. This can be problematic if a library administrator accidentally shares a list. 
[Notion](https://www.notion.so/lyrasis/Allow-users-to-unshare-a-list-in-the-Collection-Manager-ddda10ea6dcd4ea699dc90205686e0ca)

This PR also has changes for generating automatic documentation through in-code annotations and request-response models.
[Pydantic](https://docs.pydantic.dev/) and [Flask-Pydantic-Spec](https://github.com/turner-townsend/flask-pydantic-spec) were used to achieve the model setup and autogeneration respectively.
The API swagger can be seen on a local setup on the `http://localhost:6500/apidoc/swagger` endpoint

## How Has This Been Tested?
Test cases have been update to test unshares

## Checklist

- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by RishiDiwanTT** [comment]
Moved this PR to `Draft` for the initial implementation of Pydantic based documentation

File: api/admin/controller/__init__.py [hunk 1]
```diff
@@ -777,6 +778,21 @@ def genres(self):
 
 
 class CustomListsController(AdminCirculationManagerController):
+    class CustomListSharePostResponse(BaseModel):
+        successes: int = 0
+        failures: int = 0
+
+    class CustomListPostRequest(BaseModel):
+        name: str
+        id: int = None
```
Comment (io7m) [inline api/admin/controller/__init__.py:787]: I think the type should be `Optional[int]`.

Comment (RishiDiwanTT) [reply api/admin/controller/__init__.py:787]: Pydantic, oddly, does not seem to cover this combination of Typed variable with a None value. It just... works as None, no problem.
I'll add the `Optional` to remove ambiguity.

**Comment by RishiDiwanTT** [comment]
@io7m bump

**Review by io7m — APPROVED** [review]
Looks good!

**Pull request merged**